doc/cug.tex

```diff
@@ -4551,6 +4489,380 @@ \subsubsection{Collapsible Families In Suite Graphs}
+when the suite is parsed), whereas dynamic cycling extends a workflow at run
+time, potentially indefinitely.
+
+Here's a dynamic cycling workflow of two-monthly model runs for one year, with
+previous-instance model dependence (e.g.\ for model restart files):
+\lstset{language=suiterc}
+\begin{lstlisting}
+[scheduling]
+    initial cycle point = 2020-01
+    final cycle point = 2020-12
+    [[dependencies]]
+        [[[R1]]]  # Run once, at the initial point.
+            graph = "prep => model"
+        [[[P2M]]]  # 2-month intervals between the initial and final points.
+            graph = "model[-P2M] => model => post_proc & archive"
+[runtime]
+    [[model]]
+        script = "run-model $CYLC_TASK_CYCLE_POINT"
+\end{lstlisting}
+And here's how to do the same thing with parameterized tasks:
+\lstset{language=suiterc}
+\begin{lstlisting}
+[cylc]
+    [[parameters]]
+        chunk = 1..6
+[scheduling]
+    [[dependencies]]
+        graph = """prep => model<chunk=1>
+                     model<chunk-1> => model<chunk> =>
+                       post_proc<chunk> & archive<chunk>"""
+[runtime]
+    [[model<chunk>]]
+        script = """
+# Compute start date from chunk index and interval, then run the model.
+INITIAL_POINT=2020-01
+INTERVAL_MONTHS=2
+OFFSET_MONTHS=(( (CYLC_TASK_PARAM_chunk - 1)*INTERVAL_MONTHS ))
+OFFSET=P${OFFSET_MONTHS}M  # e.g. P4M for chunk=3
+run-model $(cylc cyclepoint --offset=$OFFSET $INITIAL_POINT)"""
+\end{lstlisting}
+
+For this simple case both suites achieve the same result. Scheduling efficiency
+is more or less the same (because dynamic cycling generates continuous
+workflows with no global cycle boundaries in cylc - see \ref{HowCylcWorks}).
+Special behaviour at the start or end of a workflow, or anywhere in between,
+can be added in both cases. In general, however, dynamic cycling is more
+flexible and powerful for the following reasons:
+\begin{myitemize}
+    \item In the dynamic case you can use ISO 8601 recurrence expressions
+        to define any date-time cycle, and cylc does the difficult date-time
+        arithmetic for you.
+    \item The static suite must be finite in extent (and not too big) because
+        every task in the entire workflow, from beginning to end, is
+        predefined.  In the dynamic case it makes no difference if the final
+        cycle point is far in the future, or if there is no final point at all.
+    \item Dynamic cycling is also better for suite visualization and monitoring.
+        \begin{myitemize}
+            \item Every task in the entire static workflow is necessarily
+                visible at once, which is a problem in large or long suites. In
+                the dynamic case you can visualize as little or as much of it as
+                you like, and at run time you'll only see cycle points that are
+                currently active.  The two workflows are shown together in
+                Figure~\ref{fig-eg2}.
```

Comment:
So, I guess I'm not clear how this is working differently. Is it that in the R1 "static" case everything is hard coded, so on evaluation for visualisation everything is drawn, whereas in the "dynamic" case the items are evaluated as you go, resulting in it only visualising where its up to in the evaluation chain?